Move atdStartup, atdUpdate to python from shell script, added logging to login container

tatanithin opened the pull request:
This pull request removes support for the ATD file and service update automation in the `labvm` directory. The changes eliminate scripts, service files, and documentation related to automatically updating lab files and services, simplifying the labvm structure and disabling automated update mechanisms.

**Removal of ATD update automation:**

* Deleted the `atdFiles.sh` script, which handled downloading, installing, and updating lab files and guides on the labvm.
* Removed the `atdFiles.service` systemd unit file, which was responsible for running the `atdFiles.sh` automation.

**Removal of service updater functionality:**

* Deleted the `atdServiceUpdater.py` script, which managed detection and updating of service files via GitHub and systemd.
* Removed the `atdServiceUpdater.service` systemd unit file, which scheduled the service updater automation.

**Documentation cleanup:**

* Removed the `readme.md` file from the `labvm` directory that described the structure and usage of the labvm update and service automation.

tatanithin committed:
Add HubSpot Cloud Function and cleanup labvm services

Added hubspot_update.py implementing a unified HubSpot handler for Google Cloud Functions, along with Dockerfile.test and deployment script for testing and deployment. Removed legacy labvm/services scripts, service files, and documentation to deprecate the old service update and CVP automation system. Introduced new nested-labvm service scripts and requirements for updated service management.

tatanithin committed:
Update atd_manager.py

tatanithin committed:
Prevent duplicate cloud logging handlers

Adds a check to ensure the cloud logging handler is not added multiple times to the root logger, avoiding duplicate log entries.

tatanithin committed:
Update atd_manager.py

tatanithin committed:
Refactor ATD services structure and update scripts

Moved atd_manager.py to utils/ and updated import paths in atdStartup.py and atdUpdate.py. Added __init__.py files for new modules. Refactored atdUpdate.sh to delegate git and sync logic to Python, removed redundant shell wrappers, and improved file sync logic for Python and utils modules. Enhanced FileManager to skip binary files during sed_replace. Deleted obsolete image and shell script files.

tatanithin committed:
Replace create_directory with ensure_directory in ATDUpdate

Changed the method from create_directory to ensure_directory when updating the utils module directory to improve idempotency and avoid errors if the directory already exists.

tatanithin committed:
Add centralized cloud logging to login container scripts

Introduces cloud logging via a new utility module (cloud_logging_utils.py) and integrates it across all major scripts in the login container. Logging now captures key operations, user actions, and error conditions for improved observability and troubleshooting. Dockerfile updated to install google-cloud-logging and copy the utility module. Existing logging and status messages are enhanced or replaced with structured cloud log entries.

tatanithin committed:
Use mkdir -p for /var/run/sshd in Dockerfile

Replaces 'mkdir' with 'mkdir -p' to ensure the /var/run/sshd directory is created without error if it already exists.

tatanithin committed:
Add flexible import for cloud_logging_utils

Added logic to import cloud_logging_utils from site-packages or parent directory, improving compatibility for both Docker and local environments.

tatanithin committed:
Remove old exam upload script and add report stub

Deleted the deprecated upload_exam_old.py script from the login service and added a new stub file lab_instance_report 2.py with a CLOUD_LOGGING_AVAILABLE flag. This prepares for future reporting functionality and cleans up unused code.

tatanithin committed:
Remove unit test UI service file

Deleted the 'nested-labvm/services/unit-test/ui' file as it is no longer needed.

tatanithin committed:
Add CVP Studio reset functionality and logging improvements

Introduces a method to perform a master reset of CloudVision Studios in ConfigureTopology.py, triggered when the 'reset' lab is selected. Refactors device selection logging in login.py for clarity. Removes unused log_operation_error imports from several scripts and fixes a typo in an error message in upload_exam_unattended.py.

rparkinarista commented on nested-labvm/atd-docker/login/Dockerfile:
```diff
@@ -11,9 +11,9 @@ RUN curl -SsL https://github.com/boxboat/fixuid/releases/download/v0.5/fixuid-0.
+RUN pip install ruamel.yaml rcvpapi cvprac pyaml paramiko scp urllib3 libvirt-python jsonrpclib-pelix google-cloud-storage cloudvision google-cloud-logging
 
-RUN mkdir /var/run/sshd
+RUN mkdir -p /var/run/sshd
```
This one caught me out too

rparkinarista commented on nested-labvm/atd-docker/login/Dockerfile:
```diff
@@ -29,6 +29,10 @@ WORKDIR /home/arista
 
+COPY src/cloud_logging_utils.py /usr/local/lib/python3.9/site-packages/
+
+RUN chmod 644 /usr/local/lib/python3.9/site-packages/cloud_logging_utils.py
```
@tatanithin what are your thoughts on this one?

rparkinarista reviewed (approved):
I just commented on a couple of the copilot items.

tatanithin committed:
Add confirmation dialog to exam submission flow

A confirmation prompt is now shown before submitting an exam in honorlock-examsubmit-v1.js to prevent accidental submissions. Dockerfile changes ensure cloud_logging_utils.py is available and executable in /usr/local/bin.

tatanithin committed:
Merge branch 'nested/2025.1' into unit-test-nithin
